src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs

```diff
@@ -52,5 +75,68 @@ private enum PowerShellProcessArchitecture
             X86,
             X64
         }
+
+        private async Task CheckPackageManagement()
+        {
+            PSCommand getModule = new PSCommand().AddCommand("Get-Module").AddParameter("ListAvailable").AddParameter("Name", "PackageManagement");
+            foreach (PSModuleInfo module in await _powerShellContextService.ExecuteCommandAsync<PSModuleInfo>(getModule))
+            {
+                // The user has a good enough version of PackageManagement
+                if(module.Version >= s_desiredPackageManagementVersion)
+                {
+                    break;
+                }
+
+                _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                var takeActionText = "Yes";
+                MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams
+                {
+                    Message = "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                    Type = MessageType.Warning,
+                    Actions = new []
+                    {
+                        new MessageActionItem
+                        {
+                            Title = takeActionText
+                        },
+                        new MessageActionItem
+                        {
+                            Title = "Not now"
+                        }
+                    }
+                });
+
+                // If the user chose "Not now" ignore it for the rest of the session.
+                if (messageAction?.Title == takeActionText)
+                {
+                    StringBuilder errors = new StringBuilder();
+                    await _powerShellContextService.ExecuteScriptStringAsync(
+                        "powershell.exe -NoLogo -NoProfile -Command 'Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber'",
```

Comment:
This no longer works because the site requires TLS 1.2. I would recommend added code to force to use TLS 1.2

`powershell.exe -NoLogo -NoProfile -Command '[Net.ServicePointManager]::SecurityProtocol=[Net.SecurityProtocolType]::Tls12; Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber`